### DISCUSSION TITLE
[parcel2] Output filename by formula? `filename: "[name].bundle"`

### DISCUSSION THREAD

**Discussion by codydjango** [parcel-bundler/parcel#6011] [Q&A]
I'm migrating from webpack where a large project has been using
```
        output: {
            filename: '[name].bundle.js',
            path: path.resolve(__dirname, BUILD_DIR), 
        }
```

Parceljs2 by default uses original filename as output filename, so to handle my migration I either need to rework the current project to drop the "bundle" insert (and hundreds of references), or write some postprocessing step where I rename the output files to match the current project.

Thanks for your insight!!! 🙏

**Comment by mischnic** [comment]
Namer plugins control the filenames of bundles.

Parcel would currently create files like `dist/index.js`, `dist/my-async-import.1afed92.js`. Do you just want them to end with `.bundle.js` instead (so `dist/my-async-import.1afed92.bundle.js`)?

**Reply by codydjango** [reply]
Thank you! Hah, we answered at the same time. Can you point me to some docs or an example I can follow?

I basically have a list of 20 js files that I want to map to 20 output files. Keep it as simple as possible, for now.  The output files I would like to have a name of [name].bundle.js.

**Comment by codydjango** [comment]
Ah, looks like this is what the "Namer" plugin is for.. but I seem to have a difficult time finding docs on how to integrate a simple plugin with parcel2?

**Reply by mischnic** [reply]
Here are the docs (forgot to link to them in the previous comment): https://v2.parceljs.org/plugin-system/namer/

An example for a third-party Parcel plugin: https://github.com/mischnic/parcel-namer-preserve-structure

**Reply by codydjango** [reply]
ah, thank you. yeah I think the missing piece was this one:
```
# .parcelrc
{
	"extends": "@parcel/config-default",
	"namers": ["@mischnic/parcel-namer-preserve-structure", "..."]
}
```
Can I use the same `"extends": "@parcel/config-default",`? this will just apply my namer to all default parcel behavior?

**Reply by mischnic** [reply]
Yes, `"extends": "@parcel/config-default",` means: "use the default config, but override/merge what with what's defined in this config"

**Reply by djviy** [reply]
**parcel-namer-preserve-structure**  doesn't work with js files. 
I also tried **[parcel-namer-rewrite](https://www.npmjs.com/package/parcel-namer-rewrite)** and **[parcel-namer-custom](https://www.npmjs.com/package/parcel-namer-custom?activeTab=readme)**.
And **parcel-namer-rewrite** can rename anything but index files. 
@mischnic Is there any way to preserve  .js name assosiated with index.html?
